app/controllers/federations_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+##
+# Provides Federated content via JSON, to be displayed within a content Collection
+class FederationsController < ApplicationController
+  # The federated action which is used to retrieve federated results via the foederati gem.
+  def show
+    @collection = find_collection
+    provider = params[:id]
+    federation_config = @collection.federation_configs.where(provider: provider).first
+    @query = "#{params[:query]} #{federation_config.context_query}"
```

Comment:
Tweaking this to: `[params[:query], federation_config.context_query].compact.join(' ')` would prevent needless spaces going into the query sent to the provider when the user enters an empty search, e.g. http://localhost:3000/portal/en/collections/music?q=&view=grid